test/upgrade/prober/sender.go

```diff
@@ -88,8 +94,33 @@ func (p *prober) deploySender() {
 func (p *prober) removeSender() {
 	p.log.Info("Remove of sender deployment: ", senderName)
 
+	foreground := metav1.DeletePropagationForeground
+	dOpts := metav1.DeleteOptions{
+		GracePeriodSeconds: pointer.Int64(0),
+		PropagationPolicy:  &foreground,
+	}
 	err := p.client.Kube.AppsV1().
 		Deployments(p.client.Namespace).
-		Delete(p.config.Ctx, senderName, metav1.DeleteOptions{})
+		Delete(p.config.Ctx, senderName, dOpts)
 	p.ensureNoError(err)
+
+	var d *appsv1.Deployment
+	pollErr := wait.PollImmediate(time.Second, time.Minute, func() (done bool, err error) {
+		// Save err and deployment for error reporting.
+		d, err = p.client.Kube.AppsV1().
+			Deployments(p.client.Namespace).
+			Get(p.config.Ctx, senderName, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			return true, nil
+		}
+		if err != nil {
+			return false, nil
```

Comment:
```suggestion
			return false, err
```

Reply:
@pierDipi Hey, I think this should be fixed before merging...

Reply:
Done